packages/plugins/src/sample-project/index.ts

```diff
@@ -23,7 +24,53 @@ export const sampleContractFileName = 'HelloSphinx.sol'
 export const sampleScriptFileName = 'HelloSphinx.s.sol'
 export const sampleTestFileName = 'HelloSphinx.t.sol'
 
-export const init = (
+const handleInit = async () => {
+  if (process.env.SPHINX_INTERNAL_TEST__SKIP_GIT === 'true') {
+    return
+  }
+
+  // Commit all changes
+  const spawnOutput = await spawnAsync('git', ['init'])
+  if (spawnOutput.code !== 0) {
+    // The `stdout` contains the trace of the error.
+    console.log(spawnOutput.stdout)
+    // The `stderr` contains the error message.
+    console.log(spawnOutput.stderr)
+    process.exit(1)
+  }
+}
+
+const handleCommit = async () => {
+  if (process.env.SPHINX_INTERNAL_TEST__SKIP_GIT === 'true') {
+    return
+  }
+
+  // Commit all changes
+  const gitAddOutput = await spawnAsync('git', ['add', '.'])
+  console.log(gitAddOutput)
+  if (gitAddOutput.code !== 0) {
+    // The `stdout` contains the trace of the error.
+    console.log(gitAddOutput.stdout)
+    // The `stderr` contains the error message.
+    console.log(gitAddOutput.stderr)
+    process.exit(1)
+  }
```

Comment:
To make the setup process smoother, and because installing our library requires a git repo, I decided to just automatically setup git and commit the changes after the init command finishes.

Reply:
Would be nice to document this in the code